src/AbstractUI.lhs

```diff
@@ -275,21 +275,26 @@ demoteLaw thnm whichL reqs
 \subsubsection{Classifying Laws}
 
 \begin{code}
-findLaw :: MonadFail m => Theory -> String -> m [Law]
-findLaw thry lnm
- | lnm == "."      =  return lws
+findLaw :: MonadFail m => Theory -> String -> REqState -> m [Law]
+findLaw thry lnm reqs
+ | lnm == "*"      =  return lwsDep
+ | lnm == "."      =  return lwsCur
  | null law1       =  fail ("law '"++lnm++"': not found")
  | otherwise       =  return [theLaw]
  where
-   lws = laws thry
-   (_,law1,_) = extract (((lnm==) . fst) .fst) lws
+   currTh = currTheory reqs
+   thys = theories reqs
+   lwsCur = laws thry
+   lwsDep = concat $ map laws (fromJust $ getTheoryDeps currTh thys)
```

Comment:
This won't work - it adds the classification into the theory where the command was issued. 
It needs to process each dependent theory indivudually - see `assumeDepConj` in `Theories.lhs`.

Reply:
I'm going to accept this anyway - the above isn't a deal breaker.

You need to focus now on applying these simplifiers in a proof.